src/libraries/System.Private.CoreLib/src/System/Runtime/Intrinsics/Wasm/WasmBase.cs

```diff
@@ -0,0 +1,198 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Runtime.CompilerServices;
+using System.Runtime.Intrinsics;
+
+namespace System.Runtime.Intrinsics.Wasm
+{
+    [Intrinsic]
+    internal abstract class WasmBase
```

Comment:
We should take this (well https://github.com/dotnet/runtime/issues/53730) to actual API review for .NET 8.

Provided that the relevant hookups exist for the xplat `Vector128<T>` (and `Vector<T>`) APIs, WASM will also be able to light up on any paths using those APIs prior to WASM support being directly exposed.

Reply:
> Provided that the relevant hookups exist for the xplat `Vector128<T>` (and `Vector<T>`) APIs, WASM will also be able to light up on any paths using those APIs prior to WASM support being directly exposed.

IIRC we already handle xplat for Vector128<T> on wasm. Vector<T> is next for me to look into, I hope I can speedup string and other parts on wasm.

Reply:
Thanks for the info!

We're looking at moving more code to use `Vector128<T>` xplat APIs where possible so that we can support more platforms with less code overall (and only using the platform specific APIs where it gives significant improvement or where functionality can't be represented using the xplat APIs).

So supporting `Vector128<T>` will allow more WASM light-up faster.